apps/ios/LiveVideo/LiveVideo/Managers/Auth/AuthManager.swift

```diff
@@ -3,21 +3,61 @@
 //
 
 import Combine
-import SwiftyUserDefaults
+import KeychainAccess
 
 class AuthManager: ObservableObject {
-    @Published var isSignedOut = Defaults.userIdentity.isEmpty
-    @Published var userIdentity = Defaults.userIdentity
+    @Published var isSignedOut = true
+    @Published var userIdentity = ""
+    private let keychain = Keychain()
+    private let userIdentityKey = "UserIdentity" // Store user identity in keychain because it is PII
+    private let passcodeKey = "Passcode" // Store passcode in keychain because it is secret
+    private var api: API!
 
-    func signIn(userIdentity: String) {
-        Defaults.userIdentity = userIdentity
-        self.userIdentity = userIdentity
-        isSignedOut = false
+    func configure(api: API) {
+        self.api = api
+
+        if let userIdentity = keychain[userIdentityKey], let passcode = keychain[passcodeKey] {
+            isSignedOut = false
+            self.userIdentity = userIdentity
+            try? configureAPI(passcode: passcode)
+        }
+    }
+    
+    func signIn(userIdentity: String, passcode: String, completion: @escaping (Result<Void, Error>) -> Void) {
+        do {
+            try configureAPI(passcode: passcode)
+        } catch {
+            completion(.failure(error))
+            return
+        }
+        
+        api.request(VerifyPasscodeRequest()) { [weak self] result in
+            guard let self = self else {
+                return
+            }
+            
+            switch result {
+            case .success:
+                self.keychain[self.userIdentityKey] = userIdentity
+                self.keychain[self.passcodeKey] = passcode
+                self.userIdentity = userIdentity
+                self.isSignedOut = false
```

Comment:
Is this called on main thread?

Reply:
yes, tested. Its called on the main thread so we are good here. worth adding a comment I guess.

Reply:
I think in the app layer it is probably common to assume completion closures are called on the main queue unless otherwise stated in order to keep concurrency as simple as possible. But to help with this I added a new comment inside `API.swift` that explains that all API responses are called on the main queue. I think it's probably best to explain this at the API level instead of the various places API requests are made.